### ISSUE TITLE
Attach files to a request

### ISSUE THREAD

**Issue by boutsen** [DalSoft/DalSoft.RestClient#7]
Is it possible to attach a file for Post() ?

You can set the Content-Type: multipart/form-data. 

If yes ? Do you have an example ?

**Comment by DalSoft** [comment]
Not yet, but something I will be adding is a working example doc of how to extend it to do anything you want. As it stands today all to need to do is implement a IHttpClientWrapper that does this.

And pass it when new'ing up the client for example:
var restClient = new RestClient(new MyHttpClientWrapperTheDoesMultipartFormData(), baseUrl);

I do have plans to make this even easier and support selection of different HttpClientWrappers based on the request or response.

**Comment by cburnette** [comment]
@DalSoft  I would love a concrete example of this in the Readme, especially using a Stream.

**Comment by DalSoft** [comment]
@boutsen @cburnette I'm currently working on a Handler

https://github.com/DalSoft/DalSoft.RestClient/blob/features/multipart-form-data-support/DalSoft.RestClient/Handlers/MultipartFormDataHandler.cs 

to do this, this should drop sometime this week. The code will look like:

```cs
dynamic restClient = new RestClient("http://en.directupload.net/index.php", new Config
(
    new MultipartFormDataHandler()
));
            
var multipartContentType = new Dictionary<string, string> { { "Content-Type", $"multipart/form-data;boundary=\"Upload----{Guid.NewGuid()}\"" } };
var filepath = Path.GetDirectoryName(GetType().GetTypeInfo().Assembly.Location) + "/DalSoft.jpg";
var fileBytes = File.ReadAllBytes(filepath);

var result = await restClient.Query(new { mode = "upload" })
.Post(new
{
    bilddatei = fileBytes, filename = "dalsoft.jpg" //bilddatei is image file in german incase you were wondering
}, 
    multipartContentType
);

Assert.That(result.HttpResponseMessage.StatusCode, Is.EqualTo(HttpStatusCode.OK));
Assert.That(result.ToString(), Does.Contain("http://www.directupload.net/file/d/"))
```

**Comment by DalSoft** [comment]
Implemented in 3.1.0
https://github.com/DalSoft/DalSoft.RestClient/wiki/Available-Plugins#multipartformdatahandler
